network/network.go

```diff
@@ -368,31 +396,45 @@ func (n *Network) connectToKnownNodes(nodeDID did.DID) error {
 	if err != nil {
 		return err
 	}
+	n.assumeNewNode = len(otherNodes) == 0
+	if n.assumeNewNode {
+		log.Logger().Infof("Assuming this is a new node, discovered NutsComm addresses are processed with a %s delay", newNodeConnectionDelay)
+	}
 	for _, node := range otherNodes {
-		if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
-			// Found local node, do not discover.
-			continue
-		}
-	inner:
-		for _, service := range node.Service {
-			if service.Type == transport.NutsCommServiceType {
-				var nutsCommUrl transport.NutsCommURL
-				if err = service.UnmarshalServiceEndpoint(&nutsCommUrl); err != nil {
-					log.Logger().
-						WithError(err).
-						WithField(core.LogFieldDID, node.ID.String()).
-						Warn("Failed to extract NutsComm address from service")
-					continue inner
-				}
+		n.connectToDID(nodeDID, node)
+	}
+
+	return nil
+}
+
+func (n *Network) connectToDID(nodeDID did.DID, node did.Document) {
+	if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
+		// Found local node, do not discover.
+		return
+	}
+inner:
+	for _, service := range node.Service {
+		if service.Type == transport.NutsCommServiceType {
+			var nutsCommUrl transport.NutsCommURL
+			if err := service.UnmarshalServiceEndpoint(&nutsCommUrl); err != nil {
 				log.Logger().
+					WithError(err).
 					WithField(core.LogFieldDID, node.ID.String()).
-					WithField(core.LogFieldNodeAddress, nutsCommUrl.Host).
-					Info("Discovered Nuts node")
-				n.connectionManager.Connect(nutsCommUrl.Host, node.ID)
+					Debug("Failed to extract NutsComm address from service")
+				continue inner
+			}
+			log.Logger().
+				WithField(core.LogFieldDID, node.ID.String()).
+				WithField(core.LogFieldNodeAddress, nutsCommUrl.Host).
+				Debug("Discovered Nuts node")
+			if n.assumeNewNode {
+				// Connect to NutsComm addresses with a delay.
+				n.connectionManager.Connect(nutsCommUrl.Host, node.ID, newNodeConnectionDelay)
```

Comment:
Maybe a bit odd, that `assumeNewNode` is set on startup; when the node has been running for a while it isn't a new node anymore; but new addresses will be 5 minutes delayed before being connected to.

Maybe `newNodeConnectionDelay` should be a fixed time, e.g. `time.Now().Add(5 * time.Minute)`, which is then substracted in the `Connect()` call here?

Then after 5 minutes, new nodes are being connected to immediately.

Or simplify it, if `boottime + 5 minutes` has passed, pass `0`

Reply:
I've applied the last suggestion.

Not sure if this actually relevant. A node that has never been restarted is an observer node. I don't think the delay is going to make a difference there.

Reply:
Sort of agree; but maybe it will work a bit nicer in workshop environments. We'll see.